uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/PathUtil.java

```diff
@@ -51,4 +69,14 @@ public String getNiogitRepoPath(org.uberfire.java.nio.file.Path path) {
         }
     }
 
+    public Optional<String> extractBranch(String uri) {
+        final Pattern pattern = Pattern.compile("^[A-Za-z]+://([^@]+)@.*");
```

Comment:
Pattern compilation is relatively expensive operation. Could we make this pattern into private static final field of the class?